src/Adapter/Shipment/CommandHandler/SplitShipmentHandler.php

```diff
@@ -31,48 +31,65 @@
 use PrestaShop\PrestaShop\Core\Domain\Shipment\CommandHandler\SplitShipmentHandlerInterface;
 use PrestaShop\PrestaShop\Core\Domain\Shipment\Exception\CannotEditShipmentShippedException;
 use PrestaShop\PrestaShop\Core\Domain\Shipment\Exception\ShipmentNotFoundException;
+use PrestaShop\PrestaShop\Core\Domain\Shipment\Service\ShipmentSplitterInterface;
 use PrestaShopBundle\Entity\Repository\ShipmentRepository;
 use PrestaShopBundle\Entity\ShipmentProduct;
-use Throwable;
+use Symfony\Contracts\Translation\TranslatorInterface;
 
-#[AsCommandHandler()]
+#[AsCommandHandler]
 class SplitShipmentHandler implements SplitShipmentHandlerInterface
 {
     public function __construct(
-        private readonly ShipmentRepository $repository,
+        private ShipmentRepository $repository,
+        private ShipmentSplitterInterface $splitter,
+        private TranslatorInterface $translator,
     ) {
     }
 
-    /**
-     * @param SplitShipment $command
-     */
     public function handle(SplitShipment $command): void
     {
+        $shipmentId = $command->getShipmentId()->getValue();
         $carrierId = $command->getCarrierId()->getValue();
-        $shipmentToRemoveProductId = $command->getShipmentId()->getValue();
-        $products = $command->getOrderDetailQuantity()->getValue();
-        $findShipment = $this->repository->findOneBy(['id' => $shipmentToRemoveProductId]);
 
-        if (!$findShipment) {
-            throw new ShipmentNotFoundException(sprintf('Could not find shipment with id "%s"', $shipmentToRemoveProductId));
+        $shipment = $this->repository->findById($shipmentId);
+
+        if (!$shipment) {
+            throw new ShipmentNotFoundException(
+                $this->translator->trans(
+                    'Could not find shipment with id "%id%".',
```

Comment:
I see that we have another string for the same purpose in this PR, we can reuse it to manage translations efficiently.

```suggestion
                    ''Shipment with id "%id%" was not found.',
```